polkadot/zombienet-sdk-tests/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+				.with_node(|node| node.with_name("honest-5"))
+				.with_node(|node| node.with_name("honest-6"))
+				.with_node(|node| node.with_name("honest-7"))
+				.with_node(|node| node.with_name("honest-8"))
+				.with_node(|node| {
+					node.with_name("malicious-disputer")
+						.with_command("malus")
+						.with_subcommand("dispute-ancestor")
+						.with_args(vec![
+							"--no-hardware-benchmarks".into(),
+							"--insecure-validator-i-know-what-i-do".into(),
+							"-lMALUS=trace".into(),
+						])
+				})
+		})
+		.with_parachain(|p| {
+			p.with_id(2000)
+				.cumulus_based(true)
+				.with_registration_strategy(RegistrationStrategy::InGenesis)
+				.with_collator(|n| n.with_name("collator").with_command("polkadot-parachain"))
+		})
+		.build()
+		.unwrap()
+		.spawn_native()
+		.await?;
+
+	Ok(network)
+}
+
+// FIXME: deduplicate this too
+pub async fn spawn_honest_network() -> Result<Network<LocalFileSystem>, Error> {
+	let network = NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			let patch = runtime_config();
+
+			r.with_chain("westend-local")
+				.with_genesis_overrides(patch)
+				.with_default_command("polkadot")
+				.with_default_args(vec![
+					"--no-hardware-benchmarks".into(),
+					"--insecure-validator-i-know-what-i-do".into(),
+					"-lparachain=debug,parachain::dispute-coordinator=trace".into(),
+				])
+				.with_node(|node| node.with_name("honest-0"))
+				.with_node(|node| node.with_name("honest-1"))
+				.with_node(|node| node.with_name("honest-2"))
+				.with_node(|node| node.with_name("honest-3"))
+				.with_node(|node| node.with_name("honest-4"))
+				.with_node(|node| node.with_name("honest-5"))
+				.with_node(|node| node.with_name("honest-6"))
+				.with_node(|node| node.with_name("honest-7"))
+				.with_node(|node| node.with_name("honest-8"))
+				.with_node(|node| node.with_name("honest-9"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2000)
+				.cumulus_based(true)
+				.with_registration_strategy(RegistrationStrategy::InGenesis)
+				.with_collator(|n| n.with_name("collator").with_command("polkadot-parachain"))
+		})
+		.build()
+		.unwrap()
+		.spawn_native()
+		.await?;
+
+	Ok(network)
+}
+
+pub async fn get_client(
+	network: &Network<LocalFileSystem>,
+	name: &str,
+) -> Result<OnlineClient<PolkadotConfig>, Error> {
+	let client = network
+		.get_node(name)?
+		.client::<subxt::config::polkadot::PolkadotConfig>()
+		.await?;
+	Ok(client)
+}
+
+pub async fn wait_for_block(
+	number: u32,
+	client: OnlineClient<PolkadotConfig>,
+) -> Result<(), Error> {
+	println!("Waiting for block #{number}:");
+	let mut best = client.blocks().subscribe_best().await?;
+
+	while let Some(block) = best.next().await {
+		let n = block?.header().number;
+		println!("Current best block: #{n}");
+		if n >= number {
+			break;
+		}
+	}
+
+	Ok(())
+}
+
+pub async fn wait_for_metric(node: &NetworkNode, metric: &str, value: u64) -> Result<(), Error> {
+	println!("Waiting for {metric} to reach {value}:");
+	loop {
```

Comment:
We should add some kind of `sleep` between to not constantly query the prometheus server of the node. I think at least 1 sec.